Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -442,12 +442,17 @@ - (void)setValuesForKeysWithDictionary:(NSDictionary*)keyedValues {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSDictionary*)dictionaryWithValuesForKeys:(NSArray*)keys {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSDictionary* results = [NSMutableDictionary dictionary];
+
+    for (NSString* key in keys) {
+        id retValue = [self valueForKey:key];
+        [results setValue:((retValue != nil) ? retValue : [NSNull null]) forKey:key];
```

Comment:
You can simplify this to `retValue ?: [NSNull null]` -- If you elave out the true-expression from a ternary, it uses the condition.